### PR TITLE
create,store,add job,category

### PR DESCRIPTION
**Pull request by esraa33-95**

### CODE REVIEW

File: app/Http/Controllers/Admin/JobController.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'title'=> 'required|string',
+            'description'=>'required|string',
+            'responsability'=> 'required|string',
+            'job_nature'=> 'required|string',
+            'location' => 'required|string',
+            'salary_from'=> 'required|numeric',
+            'salary_to'=> 'required|numeric',
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/Admin/JobController.php:43]: you can validate `salary_to` is greater than `salary_from`, if you say in front end that expected salary from 1000 till 800 user will be confused

File: app/Http/Controllers/Admin/JobController.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            'qualification'=> 'required|string',
+             'date_line' => 'required|date',
+             'published' => 'boolean',
+             'category_id'=> 'required|integer|exists:categories,id',
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/Admin/JobController.php:47]: i love this rule

Comment (Nora-Lootfy) [reply app/Http/Controllers/Admin/JobController.php:47]: makes me happy😂

File: lang/ar/jobs.php
Comment (Nora-Lootfy) [inline lang/ar/jobs.php:1]: oh you have make it multilingual, nice😊

File: resources/views/admin/add_category.blade.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  <main>
+    <div class="container my-5">
+      <div class="bg-light p-5 rounded">
+        <h2 class="fw-bold fs-2 mb-5 pb-2">Add Category</h2>
+        <form action="{{route('jobs.store')}}" method="post" class="px-md-5" enctype="multipart/form-data">
+            @csrf
+          <div class="form-group mb-3 row">
+            <label for="" class="form-label col-md-2 fw-bold text-md-end">Category name:</label>
+            <div class="col-md-10">
+              <input type="text" placeholder="" class="form-control py-2" name="category_name" />
```
Comment (Nora-Lootfy) [inline resources/views/admin/add_category.blade.php:32]: why there is no `@error` or old values printed

File: resources/views/errors/404.blade.php
Comment (Nora-Lootfy) [inline resources/views/errors/404.blade.php:1]: perfect

File: resources/views/public/contact.blade.php [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
         <!-- Header End -->
-        @include('public.includes.about-header')
+        @include('public.includes.header')
         <!-- Header End -->
- @endsection
  
- @section('page')
      
         <!-- Contact Start -->
         @include('public.includes.contact1')
```
Comment (Nora-Lootfy) [inline resources/views/public/contact.blade.php:13]: still `contact1` i don't prefer it but as you like

File: routes/web.php [hunk 5]
```diff
@@ -1,7 +1,13 @@
 <?php
 
+
 use Illuminate\Support\Facades\Route;
+use App\Http\controllers\Admin\JobController;
```
Comment (Nora-Lootfy) [inline routes/web.php:5]: make C in `controllers` capital, this can cause bug

File: app/Http/Controllers/PublicController.php
Comment (Nora-Lootfy) [inline app/Http/Controllers/PublicController.php:1]: you forget to remove 
```php
 public function error()
    {
        return view('public.404');
    }
```

no need for it, 404 is managed by laravel automatically

**Review by Nora-Lootfy — CHANGES_REQUESTED** [review]
Thank you all for your hard work. I just need a few small tweaks to make everything more consistent.

**Comment by Nora-Lootfy** [comment]
can you resolve the conflict our need some help, DM me if you want support

File: resources/views/admin/add_category.blade.php
Comment (engsarakhaled) [inline resources/views/admin/add_category.blade.php:1]: check <form action="{{route('jobs.store')}}" method="post" class="px-md-5" enctype="multipart/form-data"> in add category is not correct

File: routes/web.php (hunk 5 above)
Comment (esraa33-95) [reply routes/web.php:5]: done

File: resources/views/admin/add_category.blade.php (hunk 3 above)
Comment (esraa33-95) [reply resources/views/admin/add_category.blade.php:32]: done

File: app/Http/Controllers/Admin/JobController.php (hunk 1 above)
Comment (engsarakhaled) [reply app/Http/Controllers/Admin/JobController.php:44]: working on it now

File: app/Http/Controllers/Admin/JobController.php (hunk 2 above)
Comment (esraa33-95) [reply app/Http/Controllers/Admin/JobController.php:48]: thanks for you

File: routes/web.php (hunk 5 above)
Comment (engsarakhaled) [reply routes/web.php:5]: done with @esraa33-95 esraa

**Comment by Nora-Lootfy** [comment]
I am a little confused ya @esraa33-95,,

The changes i have requested, where you push it

File: app/Http/Controllers/Admin/CategoryController.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            $data['image'] = $this->uploadFile($request->image,'assets/img');
+        }
+       Category::create($data);
+       return redirect()->route('categories.index');
+    }
+
+    /**
+     * Display the specified resource.
+     */
+    public function show(string $id)
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/Admin/CategoryController.php:53]: you can remove this if you are not planning to make this page

File: app/Http/Controllers/Admin/CategoryController.php [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        $category = Category::findOrfail($id);
+        return view('admin.edit_category',compact('category'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     */
+    public function update(Request $request, string $id)
+    {
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/Admin/CategoryController.php:71]: I have just formatted your code so it looks neat

File: routes/web.php [hunk 8]
```diff
@@ -19,3 +26,56 @@
     Route::get('job-details','jobdetails')->name('jobdetails');
     
 });
+
+
+//admin
+Route::group(
+    [
+        'prefix' => LaravelLocalization::setLocale(),
```
Comment (Nora-Lootfy) [inline routes/web.php:34]: you are planning to make your website multilingual if not no need for mcamara

**Comment by Nora-Lootfy** [comment]
I have resolved the conflict for you 

@esraa33-95 
run this command so you can take the updates
```sh
git pull origin test1
```

- don't forget to migrate:fresh, i have modified description field to be text not a string
- i have formatted your code a little bit
- the index page contains a lot of details so i remove some of them, just show the details in details page not in index, index show an overview only
- test the code then DM me to merge the pull request

**Comment by engsarakhaled** [comment]
There are 2 other branches

File: app/Http/Controllers/Admin/CategoryController.php (hunk 6 above)
Comment (engsarakhaled) [reply app/Http/Controllers/Admin/CategoryController.php:54]: Check the last edited branch

**Pull request merged**